tecken/api/views.py

```diff
@@ -614,9 +622,21 @@ def upload_files(request):
             "time": {"average": time_avg},
         }
     }
+    return aggregates
+
+
+def _upload_files_content(page, qs):
+    files = []
+    batch_size = settings.API_FILES_BATCH_SIZE
+    start = (page - 1) * batch_size
+    end = start + batch_size + 1
+    has_next = False
 
     upload_ids = set()
-    for file_upload in qs.order_by("-created_at")[start:end]:
+    file_uploads = qs.order_by("-created_at")[start:end]
+    if len(file_uploads) == batch_size + 1:
+        has_next = True
+    for file_upload in file_uploads[:batch_size]:
```

Comment:
I think line 637 triggers a sql query and then this line triggers a sql query. If we're trying to optimize sql queries out of this, then maybe there's a better way to determine has_next? Like maybe you could do something like:

```
for i, file_upload in enumerate(file_uploads):
    ...
    if i > batch_size:
        has_next = True
```

Reply:
right! Fixing that